src/http/handlers/shell.ts

```diff
@@ -93,9 +76,7 @@ export async function handleShell(req: Request, res: Response, next: NextFn): Pr
         res.write(chunk);
       }
 
-      outputEnded = true;
-      errorEnded = true;
-      checkEnd();
+      res.end();
```

Comment:
We always want to wait for the process to exit and send the status code.

Currently the response would end too early and never include the status code.